File: 504-Base7/504-Base7.java
// Last updated: 7/30/2025, 12:23:46 PM
class Solution {
    public String convertToBase7(int num) {
        if(num==0){
            return "0";
        }
        StringBuilder ans = new StringBuilder();
        int temp = Math.abs(num);
        while(temp>0){
            ans.append(temp%7 + "");
            temp /= 7;
        }
        if(num<0){
            ans.append("-");
        }
        return ans.reverse().toString();
    }
}

File: 504-Base7/504-Base7.py
# Last updated: 7/30/2025, 12:23:50 PM
class Solution:
    def convertToBase7(self, num: int) -> str:
        ans = ""
        temp = abs(num)
        while(temp>0):
            ans = str(temp%7) + ans
            temp //= 7
        if num==0 :
            return "0"
        elif num>0:
            return ans
        else:
            return "-" + ans

        